viper/parser/stmt.py

```diff
@@ -420,3 +402,28 @@ def increment_dynamic_offset(dynamic_spot):
                                         typ=None, pos=getpos(self.stmt))
         else:
             raise TypeMismatchException("Can only return base type!", self.stmt)
+
+    def get_target(self, target):
+        if isinstance(target, ast.Subscript) and self.context.in_for_loop:  # Check if we are doing assignment of an iteration loop.
+            raise_exception = False
+            if isinstance(target.value, ast.Attribute):
+                list_name = "%s.%s" % (target.value.value.id, target.value.attr)
+                if list_name in self.context.in_for_loop:
+                    raise_exception = True
+
+            if isinstance(target.value, ast.Name) and \
+               target.value.id in self.context.in_for_loop:
+                list_name = target.value.id
+                raise_exception = True
+
+            if raise_exception:
+                raise StructureException("Altering list '%s' which is being iterated!" % list_name, self.stmt)
+
+        if isinstance(target, ast.Name) and target.id in self.context.forvars:
+            raise StructureException("Altering iterator '%s' which is in use!" % target.id, self.stmt)
+        target = Expr.parse_variable_location(target, self.context)
+        if target.location == 'storage' and self.context.is_constant:
+            raise ConstancyViolationException("Cannot modify storage inside a constant function: %s" % target.annotation)
+        if not target.mutable:
+            raise ConstancyViolationException("Cannot modify function argument: %s", target.annotation)
```

Comment:
you want "Cannot modify function argument: %s" %  target.annotation ('%' instead of ',')